driver-scala/src/integration/scala/org/mongodb/scala/syncadapter/SyncMongoCollection.scala

```diff
@@ -473,6 +473,31 @@ case class SyncMongoCollection[T](wrapped: MongoCollection[T]) extends JMongoCol
   override def drop(clientSession: ClientSession, dropCollectionOptions: DropCollectionOptions): Unit =
     wrapped.drop(unwrap(clientSession), dropCollectionOptions).toFuture().get()
 
+  override def createSearchIndex(indexName: String, definition: Bson) =
+    wrapped.createSearchIndex(indexName, definition).toFuture().get()
+
+  override def createSearchIndex(definition: Bson) =
+    wrapped.createSearchIndex(definition).toFuture().get()
+
+  override def createSearchIndexes(searchIndexModels: java.util.List[SearchIndexModel]) =
+    throw new UnsupportedOperationException()
```

Comment:
Is this unsupported in the reactive streams driver?

If not you can convert a java List to a scala list by doing:

```
import scala.collection.JavaConverters._
...

wrapped.createSearchIndexes(searchIndexModels.asScala.toSeq)
```

Reply:
Thanks for the suggestion! The reactive streams driver supports this. 

Since we extend JMongoCollection, the overridden methods need to align with Java's `util.Java.List` type. To compile successfully, I changed it to using: `wrapped.createSearchIndexes(searchIndexModels.asScala.toList).toFuture().get().asJava`

I noticed the [createIndexes](https://github.com/mongodb/mongo-java-driver/blob/37d275c65d53d7aa102e8db7d82b9f88c894c157/driver-scala/src/integration/scala/org/mongodb/scala/syncadapter/SyncMongoCollection.scala#L512) method in the same class, also throws UnsupportedException. I'm not a Scala expert and was curious if this is related to overloading issues. Your thoughts on this would be helpful.